plugins/inputs/logfile/logfile_test.go

```diff
@@ -417,7 +417,9 @@ func createWriteRead(t *testing.T, prefix string, logFile *LogFile, done chan bo
 	logSrc, evts := getLogSrc(t, logFile)
 	defer (*logSrc).Stop()
 	defer close(evts)
-	const numLines int = 1000
+	// Choose a large enough number of lines so that even high-spec hosts will not
+	// complete receiving logEvents before the 2nd createWriteRead() goroutine begins.
+	const numLines int = 100000
```

Comment:
I ran this test 50 times on my macbook and on a high spec EC2 instance (m5.2xlarge) running AL2:
```
go test ./plugins/inputs/logfile -run TestLogsFileAutoRemoval -count=50 -v
```

Example output from my macbook:
```
=== RUN   TestLogsFileAutoRemoval
    logfile_test.go:376: Created temp file, /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/file_auto_removal2578471407
2022/05/16 08:39:48 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/file_auto_removal2578471407 does not exist: stat : no such file or directory
    logfile_test.go:397: Fill temp file with sufficient lines to be read.
2022/05/16 08:39:48 W! [inputs.tail] Stopping tail as file no longer exists: /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/file_auto_removal3828958604
    logfile_test.go:430: Verify every line written to the temp file is received.
    logfile_test.go:376: Created temp file, /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/file_auto_removal1902835504
    logfile_test.go:444: Verify child completed.
2022/05/16 08:39:49 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/file_auto_removal1902835504 does not exist: stat : no such file or directory
    logfile_test.go:397: Fill temp file with sufficient lines to be read.
    logfile_test.go:430: Verify every line written to the temp file is received.
    logfile_test.go:447: Completed before timeout (as expected)
    logfile_test.go:451: Verify 1st temp file was auto deleted.
2022/05/16 08:39:49 W! [logfile] Failed to auto remove file /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/file_auto_removal1902835504: remove /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/file_auto_removal1902835504: no such file or directory
    logfile_test.go:466: Verify 1st tmp file created and discovered.
    logfile_test.go:469: Completed before timeout (as expected)
--- PASS: TestLogsFileAutoRemoval (1.33s)
```

And on my AL2 EC2 instance:
```
=== RUN   TestLogsFileAutoRemoval
    logfile_test.go:376: Created temp file, /tmp/file_auto_removal890554952
2022/05/16 13:40:35 The state file  for /tmp/file_auto_removal890554952 does not exist: stat : no such file or directory
    logfile_test.go:397: Fill temp file with sufficient lines to be read.
    logfile_test.go:428: Verify every line written to the temp file is received.
2022/05/16 13:40:35 W! [inputs.tail] Stopping tail as file no longer exists: /tmp/file_auto_removal235327309
    logfile_test.go:376: Created temp file, /tmp/file_auto_removal076276999
2022/05/16 13:40:35 The state file  for /tmp/file_auto_removal076276999 does not exist: stat : no such file or directory
    logfile_test.go:442: Verify child completed.
    logfile_test.go:397: Fill temp file with sufficient lines to be read.
    logfile_test.go:428: Verify every line written to the temp file is received.
    logfile_test.go:445: Completed before timeout (as expected)
    logfile_test.go:449: Verify 1st temp file was auto deleted.
2022/05/16 13:40:36 W! [logfile] Failed to auto remove file /tmp/file_auto_removal076276999: remove /tmp/file_auto_removal076276999: no such file or directory
    logfile_test.go:464: Verify 1st tmp file created and discovered.
    logfile_test.go:467: Completed before timeout (as expected)
--- PASS: TestLogsFileAutoRemoval (1.03s)
```